tests/Unit/inc/ThirdParty/Hostings/HostResolver/getHostService.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace WP_Rocket\Tests\Unit\inc\ThirdParty\Hostings\HostResolver;
+
+use Brain\Monkey\Functions;
+use WP_Rocket\ThirdParty\Hostings\HostResolver;
+use WP_Rocket\Tests\Unit\TestCase;
+
+/**
+ * @covers \WP_Rocket\ThirdParty\Hostings\HostResolver::get_host_service
+ * @uses   ::rocket_get_constant
+ *
+ * @group  Hostings
+ * @group  ThirdParty
+ */
+class Test_GetHostResolver extends TestCase {
+	public function tearDown() {
+		parent::tearDown();
+
+		unset( $_SERVER['cw_allowed_ip'] );
+		putenv( 'SPINUPWP_CACHE_PATH=' );
+	}
+
+	/**
+	 * @dataProvider configTestData
+	 */
+	public function testShouldReturnExpectedValue( $expected ) {
+		switch( $expected ) {
+			case 'cloudways':
+				$_SERVER['cw_allowed_ip'] = true;
+				break;
+			case 'pressable':
+				Functions\when( 'rocket_get_constant' )
+					->justReturn( true );
```

Comment:
Need to qualify this with which  constant to get. Else, it'll override all mocks.

Reply:
Done in https://github.com/wp-media/wp-rocket/pull/2874/commits/ffc56472c788b4d609fe3ef753af420a16d3ac44